sound/soc/sof/intel/hda-loader.c

```diff
@@ -472,6 +472,53 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
 
+/*
+ * post fw run operations for ICL,
+ * Core 3 will be powered up and in stall when HPRO is enabled
+ */
+int hda_dsp_post_fw_run_icl(struct snd_sof_dev *sdev)
+{
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int ret;
+
+	if (sdev->first_boot) {
+		ret = hda_sdw_startup(sdev);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: could not startup SoundWire links\n");
+			return ret;
+		}
+	}
+
+	hda_sdw_int_enable(sdev, true);
+
+	/*
+	 * The recommended HW programming sequence for ICL is to
+	 * power up core 3 and keep it in stall if HPRO is enabled.
+	 * Major difference between ICL and TGL, on ICL core 3 is managed by
+	 * the host whereas on TGL it is handled by the firmware.
+	 */
+	if (!hda->clk_config_lpro) {
+		int icl_core_mask = 1 << BIT(3);
```

Comment:
you can remove this variable and directly use GENMASK in the calls below maybe?

Reply:
I tried to use GENMASK(3,3) but there are multiple places to replace. I think define a variable and reuse it below is good idea.

Reply:
sorry I meant BIT(3) not 1 << BIT(3)

Reply:
Yes, this was in between checking-in. I figured out it became high value, fixed to 1 << 3.
I will remove the variable and use BIT(3) only.